### PR TITLE
Add more tests, update changelog

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

https://github.com/prettier/prettier/pull/11778#discussion_r744426191 from @fisker 

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Don't merge, I found something else.

**Comment by fisker** [comment]
I found there is a similar message in babel codebase https://github.com/babel/babel/blob/3c0152a217bfaaa9632a86ef8d3ffd59fb0d0c1f/packages/babel-parser/src/parser/error-message.js#L159.

`SloppyFunction` and `StrictFunction` :smile:

I checked this in git blame, seems the following code will trigger this.

```js
while (false) function foo(){}
```

https://github.com/babel/babel/tree/9aa3d125b27a17f6270134f190f8b40aeadd6ea7/packages/babel-parser/test/fixtures/core/categorized/invalid-fn-decl-inside-loop


This didn't throw in Prettier, but maybe we can add it to avoid regression in future.

**Comment by fisker** [comment]
`if` is checked in this line https://github.com/babel/babel/blob/d5ba35586754da12ae406f6012b24d7f02ac17a6/packages/babel-parser/src/parser/statement.js#L276

Let's add 

```js
if (false) function foo(){}
```

too.

**Comment by sosukesuzuki** [comment]
@fisker Sorry, I could not understand what your comment means. Should we add `SloppyFunction` to ignored errors in this PR?

**Comment by fisker** [comment]
> Should we add SloppyFunction to ignored errors in this PR?

I don't know, that code triggers `SloppyFunction` error didn't appear when running Prettier, I'm saying we can add those two tests to prevent regression in future.

**Comment by fisker** [comment]
I'll add them.

**Comment by fisker** [comment]
Somehow, the babel parser didn't throw, but other parsers throws.

**Review by fisker — APPROVED** [review]

**Comment by sosukesuzuki** [comment]
> Good to you?

Yes, thank you!

File: tests/format/js/sloppy-mode/function-declaration-in-lf.js [hunk 1]
```diff
@@ -0,0 +1 @@
+if (false) function foo(){}
```
Comment (fisker) [inline tests/format/js/sloppy-mode/function-declaration-in-lf.js:1]: Oops, the filename should be `if`, not `lf`...

**Pull request merged**